client/state/selectors/is-site-using-full-site-editing.js

```diff
@@ -0,0 +1,24 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import getRawSite from 'state/selectors/get-raw-site';
+import { hasStaticFrontPage } from 'state/sites/selectors';
+
+/**
+ * Checks if a site is using the new Full Site Editing experience
+ * @param {Object} state  Global state tree
+ * @param {Object} siteId Site ID
+ * @return {Boolean} True if the site is using Full Site Editing, otherwise false
+ */
+export default function isSiteUsingFullSiteEditing( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	const isActive = get( site, 'is_fse_active', false );
+	return isActive && hasStaticFrontPage( state, siteId );
```

Comment:
Do we also want to gate on static front page? Folks are still able to pick a blog page with this value. I'll add a card for blog page support. (If we pick a page as a blog page, it ignores any post_content in it)

Reply:
Added https://github.com/Automattic/wp-calypso/issues/35107 as a follow up